Library/Homebrew/requirements/java_requirement.rb

```diff
@@ -103,7 +103,7 @@ def oracle_java_os
   end
 
   def satisfies_version(java)
-    java_version_s = Utils.popen_read("#{java} -version 2>&1")[/1.\d/]
+    java_version_s = Utils.popen_read("#{java} -version 2>&1")[/(1|9)\.\d/]
```

Comment:
why not `/\d+.\d/`?

Reply:
I assumed it was hardcoded for some desirable reason, to be honest. If there's not a known one I have no objections to that.